tests/integration/base.py

```diff
@@ -444,8 +445,8 @@ def create_devices(klass):
         klass.dev_w_client = devs[1]
 
         # don't let NM trample over our fake AP
-        with open('/run/NetworkManager/conf.d/test-blacklist.conf', 'w') as f:
-            f.write('[main]\nplugins=keyfile\n[keyfile]\nunmanaged-devices+=nptestsrv,%s\n' % klass.dev_w_ap)
+        with open('/etc/NetworkManager/conf.d/99-test-denylist.conf', 'w') as f:
```

Comment:
Out of curiosity, why the change from /run to /etc ?

Reply:
This is because the changes in /run would be overwritten by our test system's /etc/NetworkManager/conf.d/90-test-ignore.conf, because /etc/ drop-in configs are read after /run/ drop-ins and the /etc/ drop-in uses a hard overwrite of `unmanaged-devices=...` in order to reset the system's default in /usr/lib/NetworkManager/conf.d/10-globally-managed-devices.conf (guess we could also downgrade the test systems static /etc/ drop-in to /run/ which should have the same effect).

So with this change we would have the following order:
1) base system: (system config, to be overwritten)
/usr/lib/NetworkManager/conf.d/10-globally-managed-devices.conf
2) setUpClass: (static test system config, i.e. management interface eth0, en*)
/etc/NetworkManager/conf.d/90-test-ignore.conf
3) create_devices: (dynamic test interfaces, like eth42, eth43, wlan1, ...)
/etc/NetworkManager/conf.d/99-test-denylist.conf

> [...] you can add additional .conf files to the /etc/NetworkManager/conf.d directory. These will be read in order, with later files overriding earlier ones. Packages might install further configuration snippets to /usr/lib/NetworkManager/conf.d. This directory is parsed first, even before NetworkManager.conf. Scripts can also put per-boot configuration into /run/NetworkManager/conf.d. This directory is parsed second, also before NetworkManager.conf. The loading of a file /run/NetworkManager/conf.d/name.conf can be prevented by adding a file /etc/NetworkManager/conf.d/name.conf. Likewise, a file /usr/lib/NetworkManager/conf.d/name.conf can be shadowed by putting a file of the same name to either /etc/NetworkManager/conf.d or /run/NetworkManager/conf.d.

https://developer.gnome.org/NetworkManager/stable/NetworkManager.conf.html